crates/swc_cli/src/commands/compile.rs

```diff
@@ -1,5 +1,220 @@
+use std::{
+    fs,
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+
+use anyhow::Context;
 use clap::Parser;
+use path_absolutize::Absolutize;
+use rayon::prelude::*;
+use relative_path::RelativePath;
+use swc::{
+    config::{Config, Options},
+    try_with_handler, Compiler, TransformOutput,
+};
+use swc_common::{sync::Lazy, FilePathMapping, SourceMap};
+use walkdir::WalkDir;
 
+/// Configuration option for transform files.
 #[derive(Parser)]
-#[clap(author, version, about, long_about = None)]
-pub struct CompileOptions {}
+pub struct CompileOptions {
+    /// Path to a .swcrc file to use
+    #[clap(long = "config")]
+    config_file: Option<PathBuf>,
+
+    /// Files to compile
+    files: Vec<PathBuf>,
+
+    /// The output directory
+    #[clap(long)]
+    out_dir: Option<PathBuf>,
+
+    /// Specify specific file extensions to compile.
+    #[clap(long)]
+    extensions: Option<Vec<String>>,
+}
+
+static COMPILER: Lazy<Arc<Compiler>> = Lazy::new(|| {
+    let cm = Arc::new(SourceMap::new(FilePathMapping::empty()));
+
+    Arc::new(Compiler::new(cm))
+});
+
+/// List of file extensions supported by default.
+static DEFAULT_EXTENSIONS: &[&str] = &["js", "jsx", "es6", "es", "mjs", "ts", "tsx"];
+
+/// Infer list of files to be transformed from cli arguments.
+/// If given input is a directory, it'll traverse it and collect all supported
+/// files.
+fn get_files_list(
+    raw_files_input: &[PathBuf],
+    extensions: &[String],
+    _include_dotfiles: bool,
+) -> anyhow::Result<Vec<PathBuf>> {
+    let input_dir = raw_files_input.iter().find(|p| p.is_dir());
+
+    let files = if let Some(input_dir) = input_dir {
+        if raw_files_input.len() > 1 {
+            return Err(anyhow::anyhow!(
+                "Cannot specify multiple files when using a directory as input"
+            ));
+        }
+
+        WalkDir::new(input_dir)
+            .into_iter()
+            .filter_map(|e| e.ok())
+            .map(|e| e.into_path())
+            .filter(|e| {
+                extensions.iter().any(|ext| {
+                    e.extension()
+                        .map(|v| v.to_str().unwrap_or("").eq(ext))
```

Comment:
`OsStr` implements `PartialEq<str>`. (https://doc.rust-lang.org/stable/std/ffi/struct.OsStr.html#impl-PartialEq%3Cstr%3E)

You can do `v == &**ext` instead.